File: решения/cs/Classes/task1379/Interval.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Classes.Task1379
{
    public class Interval
    {
        public int a;
        public int b;

        public void Slide(int value)
        {
            if (a > b)
            {
                throw new InvalidOperationException("Корректность интервала была нарушена");
            }
            this.a += value;
            this.b += value;
        }

        public String Print()
        {
            if (a > b)
            {
                throw new InvalidOperationException("Корректность интервала была нарушена");
            }
            return "[" + a + ";" + b + "]";
        }
    }
}

File: решения/java/src/Lab14/task1379/Interval.java
package Lab14.task1379;

/**
 * @author akryukov
 *         17.07.2017
 */
public class Interval {
    public int a;
    public int b;

    public void slide(int value){
        if(a > b){
            throw new IllegalStateException("Корректность интервала была нарушена");
        }
        this.a += value;
        this.b += value;
    }

    public String print(){
        if(a > b){
            throw new IllegalStateException("Корректность интервала была нарушена");
        }
        return "[" + a + ";" + b + "]";
    }
}
